### ISSUE TITLE
numpy 1.21 ufunc reduce segfault

### ISSUE THREAD

**Issue by sklam** [numba/numba#7423]
Failing test: `numba.tests.npyufunc.test_ufunc.TestUFuncs.test_ufunc_attributes`

```pytb
Fatal Python error: Segmentation fault

Current thread 0x000000010dbdbe00 (most recent call first):
  File "/path/to/numba/numba/tests/npyufunc/test_ufunc.py", line 35 in ufunc_reduce
  File "/path/to/numba/numba/tests/npyufunc/test_ufunc.py", line 58 in _test_ufunc_attributes
  File "/path/to/numba/numba/tests/npyufunc/test_ufunc.py", line 74 in test_ufunc_attributes
```

**Comment by sklam** [comment]
Related to https://github.com/numba/numba/issues/7175

**Comment by stuartarchibald** [comment]
Closed by #7449 via https://github.com/numba/numba/pull/7483.
